algebird-core/src/main/scala/com/twitter/algebird/Scan.scala

```diff
@@ -0,0 +1,361 @@
+package com.twitter.algebird
+
+import scala.collection.compat._
+import scala.collection.generic.CanBuildFrom
+
+object Scan {
+  /**
+    * Most consumers of Scan don't care about the type of the {{State}} type variable. But for those that do,
+    * we make an effort to expose it in all of our combinators.
+    * @tparam I
+    * @tparam S
+    * @tparam O
+    */
+  type Aux[-I, S, +O] = Scan[I, O] { type State = S }
+
+  implicit def applicative[I]: Applicative[({ type L[O] = Scan[I, O] })#L] = new ScanApplicative[I]
```

Comment:
you can use Kind projector here?

Reply:
I don't think we use that plugin, we could, but I'd rather not add a build dependency for the few cases where we have done this. This is mostly a demonstration anyway, since I expect if you really want a `Applicative[Scan[I, ?]]` you will want a cats implementation, but anyway, this shows you how to make one.